File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.ts
function minimumCardPickup(cards: number[]): number {
    const map: Map<number, number> = new Map();
    let ret: number = Number.MAX_VALUE;
    
    for(let i=0; i < cards.length; i++){
        if(map.has(cards[i])) ret = Math.min(ret, i-map.get(cards[i])+1);
        map.set(cards[i], i);
    }
    
    return (ret === Number.MAX_VALUE) ? -1 : ret;
};

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.cs
public class Solution {
    public int MinimumCardPickup(int[] cards) {
        var dic = new Dictionary<int, int>();
        int ret = int.MaxValue;
        //int minValue = int.MaxValue;
        
        for(int i=0; i < cards.Length; i++)
        {
            if(dic.ContainsKey(cards[i])) ret = Math.Min(ret, i-dic[cards[i]]+1);
            dic[cards[i]] = i;
        }
        
        return (ret == int.MaxValue) ? -1 : ret;
        
    }
}



